addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js

```diff
@@ -963,6 +1047,43 @@ X[]
                         contentAfter: '<p>[]abc</p>',
                     });
                 });
+                it('should ignore ZWS', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: '<p>ab\u200B[]c</p>',
+                        stepFunction: deleteBackward,
+                        contentAfter: '<p>a[]c</p>',
+                    });
+                });
+                it('should keep inline block', async () => {
+                    await testEditor(BasicEditor, {
+                        contentBefore: '<div><p>ab</p><br><span>c[]</span></div>',
+                        stepFunction: deleteBackward,
+                        contentAfter: '<div><p>ab</p><br>[]</div>',
+                    });
+                    await testEditor(BasicEditor, {
+                        contentBefore: '<div><p>cd</p><br><span>e[]</span></div>',
+                        stepFunction: async editor => {
+                            await deleteBackward(editor);
+                            await insertText(editor, 'x');
+                        },
+                        contentAfter: '<div><p>cd</p><br><span>[]x</span></div>',
```

Comment:
I guess this kinda answer my question... but how comes we don't see the span in the previous test ? Because of cleanForSave ?

Reply:
yes, because clean for save.

We indeed keep an empty inline tag after the clean for save, if it have a class.
I'll add a more explicit test about that.